##Условие:

1. Из колоды в 52 карты извлекаются случайным образом 4 карты.

  a) Найти вероятность того, что все карты – крести.

  б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.
2. На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно. Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?
3. В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все извлеченные детали окрашены?
4. В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?

### **Задание №1**
a) В колоде из 52 листов *(52/4)*  13 карт одной масти.

Воспользуемся формулой подсчета количества сочетаний по k элементов из множества n $C_n^k  = \frac{n!}{k!(n-k)!}$

И классической формулой вероятности: $P(A) = \frac{m}{n}$



In [4]:
from math import factorial
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))


Вычислим m:

In [5]:
m=combinations(13, 4)
print(f'm = {m}')

m = 715


Вычислим n:

In [6]:
n=combinations(52, 4)
print(f'n = {n}')

n = 270725


Найдем вероятность карт крести:

In [7]:
P=m/n
print(f'P(4 крести) = {round(P,4)}')

P(4 крести) = 0.0026


б) в этом случае так же общее число исходов $n = C_{52}^{4}$

а $m$
 будет складываться из 4 вариантов:

1. из 4-х карт 1 туз:
число способов достать 1 туз из 4-х $C_4^1$
, число способов достать три остальные карты $C_{48}^{3}$
, общее количество сочетаний 1 туз и 3 не туза
$C_4^1$ * $C_{48}^{3}$
2.из 4-х карт 2 туза:
число способов достать 2 туза из 4-х $C_4^2$
, число способов достать две остальные карты $C_{48}^{2}$
, общее количество сочетаний 2 туза и 2 не туза
$C_4^2$ * $C_{48}^{2}$
3.из 4-х карт 3 туза:
число способов достать 3 туза из 4-х $C_4^3$
, число способов достать оставшуюся карту $C_{48}^{1}$
, общее количество сочетаний 3 туза и 1 не туз
$C_4^3$ * $C_{48}^{1}$
4.из 4-х карт 4 туза:
число способов достать 4 туза из 4-х $C_4^4 = 1$,
общее число благоприятных исходов-сумма этих 4 вариантов.

Вычислим:

In [ ]:
n=combinations(52, 4)
print(f'n = {n}')

n = 270725


In [ ]:
m=sum([combinations(4,1)*combinations(48,3),combinations(4,2)*combinations(48,2),combinations(4,3)*combinations(48,1),1])
print(f'm = {m}')

m = 76145


Найдем вероятность P(хотя бы один туз):

In [ ]:
P=m/n
print(f'P(хотя бы 1 туз) = {round(P,4)}')

P(хотя бы 1 туз) = 0.2813


## Итоговый ответ:
а)Вероятность, что все карты крести = **0.26%**

б)Вероятность, что среди 4-х карт окажется хотя бы один туз = **28.13%**

### **Задание** №2

На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно. Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?

Для формулы вероятности $P(A) = \frac{m}{n}$ общее число исходов будет определяться количиством сочетаний 3-х элементов из 10 $n = C_{10}^{3}$, а количество благоприятных исходов
$m = 1$ -замок открылся. Вычислим $P(A)$
:

In [8]:
P=1/combinations(10,3)
print(f'P(первая попытка) = {round(P,4)}')

P(первая попытка) = 0.0083


## Итоговый ответ:
Вероятность, что замок откроется с первой попытки = **0,83**%

### **Задание** №3

В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все извлеченные детали окрашены?

Общее количество исходов будет определяться количеством сочетаний 3 деталей из 15
, а количество благоприятных исходов - количеством сочетаний 3-х окрашенных деталей из 9

Вычислим $P(3$  из  $3х)$:

In [10]:
P=combinations(9,3)/combinations(15,3)
print(f'P(3 из 3-х) = {round(P,4)}')

P(3 из 3-х) = 0.1846


## Итоговый ответ:
Вероятность, что 3 детали из 3х будут окрашены = **18,46%**

### **Задание** №4

В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?

In [11]:
P=1/combinations(100,2)
print(f'P(оба выигрышных = {round(P,4)})')

P(оба выигрышных = 0.0002)


## Итоговый ответ:
Вероятность, что оба билета билета будут выгрышные = **0,02%**